xilem/src/view/flex.rs

```diff
@@ -394,7 +489,221 @@ impl<State, Action> View<State, Action, ViewCtx> for FlexSpacer {
+    Action: 'static,
+    V: WidgetView<State, Action>,
+{
+    /// Turns this [`FlexItem`] into an [`AnyFlexChild`]
+    ///
+    /// # Examples
+    /// ```
+    /// use xilem::view::{flex, flex_item, label};
+    /// # use xilem::{WidgetView};
+    ///
+    /// # fn view<State: 'static>() -> impl WidgetView<State> {
+    /// flex(flex_item(label("Industry"), 4.0).into_any_flex())
+    /// # }
+    ///
+    /// ```
+    pub fn into_any_flex(self) -> AnyFlexChild<State, Action> {
+        AnyFlexChild::Item(flex_item(Box::new(self.view), self.params))
+    }
+}
+
+#[doc(hidden)] // Implementation detail, public because of trait visibility rules
+pub struct AnyFlexChildState<State: 'static, Action: 'static> {
+    /// Just the optional view state of the flex item view
+    #[allow(clippy::type_complexity)]
+    inner: Option<
+        <FlexItem<Box<AnyWidgetView<State, Action>>, State, Action> as View<
+            State,
+            Action,
+            ViewCtx,
+        >>::ViewState,
+    >,
+    /// The generational id handling is essentially very similar to that of the `Option<impl ViewSequence>`,
+    /// where `None` would represent a Spacer, and `Some` a view
+    generation: u64,
+}
+
+impl<State, Action> View<State, Action, ViewCtx> for AnyFlexChild<State, Action>
+where
+    State: 'static,
+    Action: 'static,
+{
+    type Element = FlexElement;
+
+    type ViewState = AnyFlexChildState<State, Action>;
+
+    fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        let gen = 0;
+        let (element, view_state) = match self {
+            AnyFlexChild::Item(flex_item) => {
+                let (element, state) = ctx.with_id(ViewId::new(gen), |ctx| flex_item.build(ctx));
+                (element, Some(state))
+            }
+            AnyFlexChild::Spacer(spacer) => {
+                // We know that the spacer doesn't need any id, as it doesn't receive or sends any messages
+                // (Similar to `None` as a ViewSequence)
+                let (element, ()) = View::<(), (), ViewCtx>::build(spacer, ctx);
+                (element, None)
+            }
+        };
+        (
+            element,
+            AnyFlexChildState {
+                inner: view_state,
+                generation: gen,
+            },
+        )
+    }
+
+    fn rebuild<'el>(
+        &self,
+        prev: &Self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        mut element: Mut<'el, Self::Element>,
+    ) -> Mut<'el, Self::Element> {
+        match (prev, self) {
+            (AnyFlexChild::Item(prev), AnyFlexChild::Item(this)) => ctx
+                .with_id(ViewId::new(view_state.generation), |ctx| {
+                    this.rebuild(prev, view_state.inner.as_mut().unwrap(), ctx, element)
+                }),
+            (AnyFlexChild::Spacer(prev), AnyFlexChild::Spacer(this)) => {
+                View::<(), (), ViewCtx>::rebuild(this, prev, &mut (), ctx, element)
```

Comment:
We should have a comment at the spacer view that it is an invariant that this view doesn't deal with messages.